### DISCUSSION TITLE
Unified RabbitMQ console for multiple clusters

### DISCUSSION THREAD

**Discussion by alphamonkey79** [rabbitmq/rabbitmq-website#2107] [General]
Wondering if this going to ever going to be a thing for on-prem installations and clusters?

**Comment by michaelklishin** [comment]
A "unified RabbitMQ console for multiple clusters" is an extremely vague problem definition. We do not guess in this community.

@alphamonkey79 this is open source software. Our management UI uses exactly the same HTTP API 3rd party tools use. Our Prometheus scraper endpoint is equally available to anyone, and so are several Grafana dashboards. You are welcome to develop such a "unified console" at any time. You can start immediately after reading this response.

RabbitMQ can be installed in 10+ different ways with different assumptions and hundreds of potential configurations,
and visualizing tens of thousands (in fact, even 1K) of different objects (e.g. queues, streams) is a true challenge that
has nothing to do with RabbitMQ or its data model or the API endpoints it offers. Our team is still trying to figure as can be seen by various Prometheus API features and Grafana dashboard visualizations. The same problem will apply to a tool that will manage N clusters.

So this is going to take a lot of effort and the result likely will be full of assumptions that won't be true even for a handful of other companies with large fleets.

In any case, RabbitMQ's license and extensibility give those who want to "ever make this a thing" enough tools to start. There are companies with thousands and tens of thousands of **clusters** (not nodes) and they found a way to operate them. I have a couple of guesses why no one has open sourced their solution: they either want a return on such an investment or their solution is entirely company-specific.

**Reply by alphamonkey79** [reply]
Thank you for the response @michaelklishin
